mono/metadata/profiler-events.h

```diff
@@ -87,3 +86,8 @@ MONO_PROFILER_EVENT_2(thread_name, ThreadName, uintptr_t, tid, const char *, nam
 MONO_PROFILER_EVENT_2(sample_hit, SampleHit, const mono_byte *, ip, const void *, context)
 
 MONO_PROFILER_EVENT_3(iomap_report, IOMap, const char *, report, const char *, old_path, const char *, new_path)
+
+MONO_PROFILER_EVENT_5(root_register, RootRegister, void *, start, size_t, size, MonoGCRootSource, kind, void *, key, const char *, msg)
+MONO_PROFILER_EVENT_1(root_unregister, RootUnregister, void *, start)
+MONO_PROFILER_EVENT_3(gc_roots, GcRoots2, int, count, void**, addresses, void**, objects)
+
```

Comment:
* I generally try to keep these events 'namespaced', so these should be `gc_roots`, `gc_root_register`, and `gc_root_unregister`.
* They should also be in the section above with the other `gc_*` events.
* `GcRoots2` -> `GCRoots`, especially since we're just removing the old event entirely.
* `gc_root_register` should be typed as `const mono_byte *start, uintptr_t size, MonoGCRootSource source, const void *key, const char *msg` for consistency with the rest of the events.
* `gc_root_unregister` should be typed as `const mono_byte *start`.
* `gc_roots` should be typed as `uint64_t count, const mono_byte *const *addresses, MonoObject *const *objects`.